test/correctness/compute_with.cpp

```diff
@@ -1222,6 +1222,495 @@ int nested_compute_with_test() {
     return 0;
 }
 
+int update_stage_test() {
+    const int f_size = 128;
+    const int g_size = 128;
+
+    Buffer<int> f_im(f_size, f_size), g_im(g_size, g_size);
+    Buffer<int> f_im_ref(f_size, f_size), g_im_ref(g_size, g_size);
+
+    {
+        Var x("x"), y("y");
+        Func f("f"), g("g");
+
+        g(x, y) = 1;
+        g(x, y) = 2 + 0 * g(x, y);
```

Comment:
I feel like it's a bit sketchy to use the "fake" dependency (`0 * g(x, y)`). Why not use 1 + g(x, y)? It would be equal to the current definition, but without this issue.

Reply:
I had it like this in the beginning, but the new tests need to check that the stages of the function are computed in the correct order, so updates can't be commutative. I updated them to just constant, but now I realized that it's actually wrong, because it's just ensures that last stage was executed last! I've updated all tests to fix that: now they are sort of building a string of N different digits in some base to make sure that stages are executed in the correct order.